app/Foliage/HackageSecurity.hs

```diff
@@ -44,8 +47,23 @@ createKeys base = do
   replicateM_ 3 $ createKey' KeyTypeEd25519 >>= writeKeyWithId (base </> "mirrors")
 
 writeKeyWithId :: FilePath -> Some Key -> IO ()
-writeKeyWithId base k =
-  writeKey (base </> keyIdString (someKeyId k) <.> "json") k
+writeKeyWithId base k = do
+  let keyId' = keyIdString $ someKeyId k
+  let publicKey' = somePublicKey k
+
+  json <- toJSON publicKey'
+  case json of
+    JSObject obj -> do
+      case lookup "keyval" obj of
+        Just (JSObject keyValObj) ->
+          case lookup "public" keyValObj of
+            Just (JSString publicKeyVal) -> do
+              printf "Generated key %s with public key: %s\n" keyId' publicKeyVal
```

Comment:
The message could be `base/keyid` or just `base` rather than just `keyid`? Not confident about what would be the most handy?

Reply:
If people are going to want to copy the public key, then perhaps we want to print it on a separate line to make it easier to copy?

Reply:
I'd say something cut&pastable like
```
  root-keys:
    3e0cce471cf09815f930210f7827266fd09045445d65923e6d0238a6cd15126f
    443abb7fb497a134c343faf52f0b659bd7999bc06b7f63fa76dc99d631f9bea1
    a86a1f6ce86c449c46666bda44268677abf29b5b2d2eb5ec7af903ec2f117a82
    bcec67e8e99cabfa7764d75ad9b158d72bfacf70ca1d0ec8bc6b4406d1bf8413
    c00aae8461a256275598500ea0e187588c35a5d5d7454fb57eac18d9edb86a56
    d4a35cd3121aa00d18544bb0ac01c3e1691d618f462c46129271bccf39f7e8ee
```

Reply:
Why decoding the json though? You have the public key already.